march_hardware_builder/src/hardware_builder.cpp

```diff
@@ -260,12 +257,43 @@ void HardwareBuilder::initUrdf()
   {
     if (!this->urdf_.initParam("/robot_description"))
     {
-      throw HardwareConfigException("Failed to load urdf from parameter server");
+      throw march::error::HardwareException(march::error::ErrorType::INIT_URDF_FAILED);
     }
     this->init_urdf_ = false;
   }
 }
 
+std::vector<march::Joint> HardwareBuilder::createJoints(const YAML::Node& joints_config) const
+{
+  std::vector<march::Joint> joints;
+  for (const YAML::Node& joint_config : joints_config)
+  {
+    const auto joint_name = joint_config.begin()->first.as<std::string>();
+    const auto urdf_joint = this->urdf_.getJoint(joint_name);
+    if (urdf_joint->type == urdf::Joint::FIXED)
+    {
+      ROS_WARN("Joint %s is fixed in the URDF, but defined in the robot yaml", joint_name.c_str());
+    }
+    joints.push_back(HardwareBuilder::createJoint(joint_config[joint_name], joint_name, urdf_joint));
```

Comment:
If the joint is fixed in the URDF, you still add it to the joint list? Won't this cause problems if the soft limits for this joint are not defined in the URDF?

Reply:
The warning is here because the joint was defined in the robots yaml, but is FIXED in the URDF. It is possible to continue. If there are no soft limits it will warn you when trying to construct the actual joint.